application/preload.js

```diff
@@ -0,0 +1,114 @@
+const { contextBridge } = require('electron');
+const fs = require('fs').promises;
+const path = require('path');
+const os = require('os');
+
+
+let baseDir;
+const APP_NAME = "Instone";
+if (process.env.NODE_ENV === 'development') {
+    baseDir = path.join(__dirname, 'mock');
+}
+else {
+    switch (process.platform) {
+        case 'win32':
+            baseDir = path.join(os.homedir(), 'AppData', 'Roaming', APP_NAME);
+            break;
+        case 'darwin':
+            baseDir = path.join(os.homedir(), 'Library', 'Application Support', APP_NAME);
+            break;
+        case 'linux':
+            baseDir = path.join(os.homedir(), '.config', APP_NAME);
+            break;
+        default:
+            throw new Error('Unsupported platform');
+    }
+}
+contextBridge.exposeInMainWorld('fileAPI', {
+    getJournals: async () => {
+        try {
+            const journals = await fs.readdir(baseDir);
+            return journals.filter(name => fs.stat(path.join(baseDir, name)).then(stats => stats.isDirectory()));
+        } catch (error) {
+            throw new Error(`Error getting journals: ${error.message}`);
+        }
+    },
+    createJournal: async () => {
+        try {
+            let journalName = 'My Journal';
```

Comment:
Will this allow us to customize the name of the journals or is this just a temp string?

If I'm understanding this correctly, it is allowing app to create the "My Journal" string, then when the user clicks on it, it will customize then it will call updateJournalName to finally update the name of the journals?

Reply:
Just updated the API. The method takes a param name!